guardian/support-frontend
# Abandoned Basket Cookie: prevent null amount

## What are you doing in this PR?

Check if amount is `NaN` and make it `'other'` if so when setting the abandoned basket cookie.



[**Trello Card**](https://trello.com/c/4m9qt4wS/463-abandoned-basket-parsing-error-in-support-frontend)

## Why are you doing this?

Since #5995 added parsing before updating the amount in the cookie we've had tons of sentry errors. It turns out that amount can be `NaN` when the cookie is set, which turns into `null` when JSON stringified. One way this can happen is if you click directly into a link that has 'other' selected. 



## How to test

Go to `uk/contribute/checkout?selected-contribution-type=ONE_OFF&selected-amount=other`, update the amount and do not see any errors in console. 



## How can we measure success?

Sentry errors go away hopefully!



## Have we considered potential risks?
